CMakeLists.txt

```diff
@@ -29,6 +29,11 @@ set (CMAKE_PROJECT_VERSION_MAJOR ${GTSAM_VERSION_MAJOR})
 set (CMAKE_PROJECT_VERSION_MINOR ${GTSAM_VERSION_MINOR})
 set (CMAKE_PROJECT_VERSION_PATCH ${GTSAM_VERSION_PATCH})
 
+# Enforce c++17 standards
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++17")
+set(CMAKE_CXX_STANDARD 17)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
```

Comment:
I think we discussed this before (with @jlblancoc ), the new recommended way to do this is `target_compile_features(XXXXXX INTERFACE cxx_std_17)`.

Reply:
Fair. I reverted this change.

Reply:
I was here. Read @ProfFan . I leave happy :100:  XD